### ISSUE TITLE
Faces are not restored properly when no theme is loaded

### ISSUE THREAD

**Issue by protesilaos** [protesilaos/spacious-padding#25]
This is an Emacs bug, which I reported here: <https://lists.gnu.org/archive/html/bug-gnu-emacs/2024-10/msg01794.html>.

**Comment by philippludwig** [comment]
Is this the reason why this package resets my colors of the line numbers whenever a new frame is opened?

**Comment by protesilaos** [comment]
> From: Philipp Ludwig ***@***.***>
> Date: Wed,  5 Feb 2025 00:53:21 -0800
>
> Is this the reason why this package resets my colors of the line
> numbers whenever a new frame is opened?

Yes, I think so.

-- 
Protesilaos Stavrou
https://protesilaos.com

**Comment by philippludwig** [comment]
Wow that was a quick reply. Do you know of any workaround? Maybe loading my custom theme before loading this package?

**Comment by protesilaos** [comment]
On 2025-02-05 11:06, Philipp Ludwig wrote:
> Wow that was a quick reply. Do you know of any workaround? Maybe
> loading my custom theme before loading this package?

Currently do you load your theme before or after enabling 
spacious-padding-mode?

Try to load the theme beforehand.

**Comment by philippludwig** [comment]
Thank you. Sadly this does not fixes it. I put spacious-padding at the very end, but it overwrites the colors of my line numbers and my modeline for some reason.

**Comment by protesilaos** [comment]
> From: Philipp Ludwig ***@***.***>
> Date: Wed,  5 Feb 2025 03:43:59 -0800
>
> Thank you. Sadly this does not fixes it. I put spacious-padding at the
> very end, but it overwrites the colors of my line numbers and my
> modeline for some reason.

Note that the line numbers are modified on purpose to remove their
background. We can discuss whether this is a good idea or not.

But the mode line backgrounds should remain intact (unless you are
configuring the the spacious-padding-subtle-mode-line).

Does your theme cover the 'mode-line', 'mode-line-active', and
'mode-line-inactive' faces?

-- 
Protesilaos Stavrou
https://protesilaos.com

**Comment by protesilaos** [comment]
> From: Philipp Ludwig ***@***.***>
> Date: Thu,  6 Feb 2025 01:09:14 -0800
>
>> Note that the line numbers are modified on purpose to remove their
>> background. We can discuss whether this is a good idea or not.

> That is a personal preference, I guess. Personally I want the line
> numbers to visual stand out, because otherwise I have sometimes some
> trouble to distinguish them from the regular text.

Fine, we will check this separately.

>> Does your theme cover the 'mode-line', 'mode-line-active', and 'mode-line-inactive' faces?
>
> Yes. The thing that "annoys" me, is that with spacious-padding enabled
> is that I get a "huge" border (in numbers: 4 pixel) around the
> modeline, and loading the theme again does not help. I have to
> explicitly call:
>
> ```
> (set-face-attribute 'mode-line-active nil
>                        :box '(:line-width 1 :color "black" :style nil))
> (set-face-attribute 'mode-line-inactive nil
>                        :box '(:line-width 1 :color "black" :style nil))
> ```
>
> But this lasts only until the next frame is created.

Have you tried to modify the 'spacious-padding-widths'? There is a
property there for the mode line width.

-- 
Protesilaos Stavrou
https://protesilaos.com

**Comment by philippludwig** [comment]
Sorry for the late reply, I was sick. 

I tried modifying `spacious-padding-widths`, but that did not work. I changed mode-line-width to 0, but nothing changed.

Here is a screenshot of the behavior: 

![Image](https://github.com/user-attachments/assets/56d98860-ad50-4025-9b59-51453be2a936)

Maybe the reason is that I am using the doom-modeline package.

**Comment by protesilaos** [comment]
> From: Philipp Ludwig ***@***.***>
> Date: Sat, 22 Feb 2025 08:56:27 -0800
>
> philippludwig left a comment (protesilaos/spacious-padding#25)
>
> Sorry for the late reply, I was sick. 

No worries! I hope you are better now.

> I tried modifying `spacious-padding-widths`, but that did not work. I
> changed mode-line-width to 0, but nothing changed.
>
> Here is a screenshot of the behavior: 
>
> ![Image](https://github.com/user-attachments/assets/56d98860-ad50-4025-9b59-51453be2a936)
>
> Maybe the reason is that I am using the doom-modeline package.

It probably is the doom-modeline which by default is taller than the
default mode line. Try without it and see how it goes.

-- 
Protesilaos Stavrou
https://protesilaos.com

**Comment by philippludwig** [comment]
Yes, you are right, it is related to doom-modeline. I think it has some custom faces, but I am no elisp expert.

Anyway, thanks for all the help!

**Comment by protesilaos** [comment]
> From: Philipp Ludwig ***@***.***>
> Date: Sat,  1 Mar 2025 03:33:09 -0800
>
> philippludwig left a comment (protesilaos/spacious-padding#25)
>
> Yes, you are right, it is related to doom-modeline. I think it has
> some custom faces, but I am no elisp expert.
>
> Anyway, thanks for all the help!

I suspect they are doing something else internally because I cannot find
a doom-modeline face related to the height of the mode line. We would
need to look into the source code.

-- 
Protesilaos Stavrou
https://protesilaos.com
